apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Tests/FilterTest.cs

```diff
@@ -48,6 +48,10 @@ public void Where_EqualTo_String()
         Assert.Equal(expected, query.ToStructuredQuery());
     }
 
+    [Fact]
+    public void Where_EqualTo_NonSerilizableValue() =>
+        Assert.Throws<ArgumentException>(() => GetEmptyQuery().Where(Filter.EqualTo("a.b", typeof(Filter))));
```

Comment:
I'd split this up slightly, to make it clear where the exception is thrown:

```csharp
// No serialization check when the filter is created.
var filter = Filter.EqualTo("a.b", typeof(Filter));
// When used as part of a query, we check that it can be serialized in that context.
Assert.Throws<ArgumentException>(() => GetEmptyQuery().Where(filter);
```